.gitattributes

```diff
@@ -0,0 +1,65 @@
+## GITATTRIBUTES
+#
+# Details per file setting:
+#   text    These files should be normalized (i.e. convert CRLF to LF).
+#   binary  These files are binary and should be left untouched.
+#
+# Note that binary is a macro for -text -diff.
+######################################################################
+
+## AUTO-DETECT
+##   Handle line endings automatically for files detected as
+##   text and leave all files detected as binary untouched.
+##   This will handle all files NOT defined below.
+* text=auto
+
+## SOURCE CODE
+*.bats     text
+*.css      text
+*.htm      text
+*.html     text
+*.js       text
+*.sh       text
+*.bash     text
+*.fish     text
+#### asdf/bin/* explicitly define as text
+*asdf      text
+*asdf-exec text
+*_asdf     text
+#### asdf/test/fixtures/* should be covered by * text=auto on L14
```

Comment:
Documentation for an event where `* text=auto` does not pick up any of these files as `text` files. It's possible, but extremely unlikely.